src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.h

```diff
@@ -158,13 +160,15 @@ class Delegate
      *   (i.e., Trickle ICE).
      *
      * @param[in] sessionId  The current session ID.
-     * @param[in] candidates A list of ICE candidate strings.
+     * @param[in] candidates A list of ICE candidate structs.
+     * Note: The callee cannot reference the `candidates` vector after this call
+     * returns, and must copy the contents over for later use, if required.
```

Comment:
The issue is not copying the vector.  The issue is that you have to _copy the fields in the structs in the vector_.  As in, https://github.com/project-chip/connectedhomeip/pull/38480/#discussion_r2089898352 was not addressed.